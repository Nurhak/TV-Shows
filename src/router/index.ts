import { createRouter, createWebHistory } from "vue-router";
import HomeView from "../views/HomeView.vue";

const routes = [
  {
    path: "/",
    name: "home",
    component: HomeView,
  },
  {
    path: "/movie/:id",
    name: "movie",
    component: () => import("../views/MovieView.vue"),
  },
];

const router = createRouter({
  history: createWebHistory(),
  routes,
});

export { routes };

export default router;
