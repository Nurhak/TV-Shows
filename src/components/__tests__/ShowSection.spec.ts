import { afterEach, describe, it, expect } from "vitest";

import { createRouter, createWebHistory } from "vue-router";
import { routes } from "@/router";
import { shallowMount } from "@vue/test-utils";
import ShowSectionVue from "../ShowSection.vue";
import { vi } from "vitest";

vi.mock("@/composables/useBreakpoints", () => {
  const useBreakpoints = vi.fn(() => {
    return {
      width: 300,
      type: "xs",
    };
  });
  return { default: useBreakpoints };
});

const router = createRouter({
  history: createWebHistory(),
  routes: routes,
});

describe("ShowSection", () => {
  afterEach(() => {
    vi.clearAllMocks();
  });

  it("should render", () => {
    const wrapper = shallowMount(
      ShowSectionVue as any,
      {
        propsData: {
          shows: [],
          genre: "test",
          isLoading: false,
        },
        global: {
          plugins: [router],
        },
      } as any
    );
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render with shows", () => {
    const wrapper = shallowMount(
      ShowSectionVue as any,
      {
        propsData: {
          shows: [{ id: 1, name: "test" }],
          genre: "test",
          isLoading: false,
        },
        global: {
          plugins: [router],
        },
      } as any
    );
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should next and back button not exist", () => {
    const wrapper = shallowMount(
      ShowSectionVue as any,
      {
        propsData: {
          shows: [],
          genre: "test",
          isLoading: false,
        },
        global: {
          plugins: [router],
        },
      } as any
    );
    expect(wrapper.find('[data-testid="next"]').exists()).toBe(false);
    expect(wrapper.find('[data-testid="previous"]').exists()).toBe(false);
  });

  it("should slide on next click", () => {
    const wrapper = shallowMount(
      ShowSectionVue as any,
      {
        propsData: {
          shows: [
            { id: 1, name: "test" },
            { id: 2, name: "test" },
            { id: 3, name: "test" },
          ],
          genre: "test",
          isLoading: false,
        },
        global: {
          plugins: [router],
        },
      } as any
    );
    const nextSpy = vi.spyOn(wrapper.vm, "next" as any);
    wrapper.find('[data-testid="next"]').trigger("click");
    expect(nextSpy).toHaveBeenCalled();
  });
});
