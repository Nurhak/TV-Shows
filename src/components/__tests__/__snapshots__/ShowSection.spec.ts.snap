// Vitest Snapshot v1

exports[`ShowSection > should render 1`] = `
"<div class=\\"flex flex-col overflow-x-auto p-2 overflow-y-hidden\\">
  <h1 class=\\"text-2xl font-bold mb-5 sticky left-0 select-none\\">test</h1>
  <!--v-if-->
  <div class=\\"flex flex-row gap-5 p-5 justify-start\\">
    <!--v-if-->
    <show-items-stub shows=\\"\\" nextclicked=\\"false\\" prevclicked=\\"false\\"></show-items-stub>
    <!--v-if-->
  </div>
</div>"
`;

exports[`ShowSection > should render with shows 1`] = `
"<div class=\\"flex flex-col overflow-x-auto p-2 overflow-y-hidden\\">
  <h1 class=\\"text-2xl font-bold mb-5 sticky left-0 select-none\\">test</h1>
  <!--v-if-->
  <div class=\\"flex flex-row gap-5 p-5 justify-start\\">
    <!--v-if-->
    <show-items-stub shows=\\"[object Object]\\" nextclicked=\\"false\\" prevclicked=\\"false\\"></show-items-stub>
    <div data-testid=\\"next\\" class=\\"flex-1 cursor-pointer flex justify-center items-center max-w-[50px] hover:bg-slate-300 hover:bg-opacity-70 transition-colors ease-in-out duration-200 rounded-sm\\">
      <chevron-right-icon-stub class=\\"w-16 h-16 fill-white\\"></chevron-right-icon-stub>
    </div>
  </div>
</div>"
`;
